# This is a sample Jupyter Notebook

Below is an example of a code cell. 
Put your cursor into the cell and press Shift+Enter to execute it and select the next one, or click 'Run Cell' button.

Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.

To learn more about Jupyter Notebooks in PyCharm, see [help](https://www.jetbrains.com/help/pycharm/ipython-notebook-support.html).
For an overview of PyCharm, go to Help -> Learn IDE features or refer to [our documentation](https://www.jetbrains.com/help/pycharm/getting-started.html).

In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

$$\text{Cálculo de: } \; \theta_1$$
$$
\theta_1 = \frac{\sum_{i=1}^n (x_i - \bar{x})(y_i - \bar{y})}{\sum_{i=1}^n (x_i - \bar{x})^2}
$$


$$\text{Cálculo de: } \; \theta_0$$
$$
\theta_0 = \bar{y} - \theta_1 \bar{x}
$$


In [7]:
dataset=pd.read_csv("models/MLE Datos.csv")
print(dataset)

     X      Y
0    1  10.06
1    2   6.60
2    3  10.91
3    4  17.96
4    5  18.47
5    6   9.09
6    7  18.80
7    8  16.39
8    9  18.59
9   10  22.64
10  11  23.58
11  12  30.82
12  13  30.04
13  14  29.49
14  15  32.78
15  16  34.33
16  17  40.98
17  18  36.18
18  19  40.25
19  20  37.58


In [10]:
X_values=dataset["X"]
Y_values=dataset["Y"]

In [12]:
print(X_values)

0      1
1      2
2      3
3      4
4      5
5      6
6      7
7      8
8      9
9     10
10    11
11    12
12    13
13    14
14    15
15    16
16    17
17    18
18    19
19    20
Name: X, dtype: int64


In [13]:
print(Y_values)

0     10.06
1      6.60
2     10.91
3     17.96
4     18.47
5      9.09
6     18.80
7     16.39
8     18.59
9     22.64
10    23.58
11    30.82
12    30.04
13    29.49
14    32.78
15    34.33
16    40.98
17    36.18
18    40.25
19    37.58
Name: Y, dtype: float64


In [16]:
# Paso 1: Cálculo de theta_1 y theta_0 usando OLS
X_mean = np.mean(X_values)
Y_mean = np.mean(Y_values)
print(X_mean)
print(Y_mean)

10.5
24.276999999999997


In [17]:
# Cálculo de theta_1
theta_1 = np.sum((X_values - X_mean) * (Y_values - Y_mean)) / np.sum((X_values - X_mean) ** 2)
print(theta_1)

1.7606165413533834


In [18]:
# Cálculo de theta_0
theta_0 = Y_mean - theta_1 * X_mean
print(theta_0)

5.790526315789471


In [19]:
# Paso 2: Verificación
X_with_const = sm.add_constant(X_values)  # Columna el intercepto
model = sm.OLS(Y_values, X_with_const).fit()
theta_0, theta_1, model.summary()

(5.790526315789471,
 1.7606165413533834,
 <class 'statsmodels.iolib.summary.Summary'>
 """
                             OLS Regression Results                            
 Dep. Variable:                      Y   R-squared:                       0.914
 Model:                            OLS   Adj. R-squared:                  0.910
 Method:                 Least Squares   F-statistic:                     192.0
 Date:                Thu, 31 Oct 2024   Prob (F-statistic):           4.82e-11
 Time:                        12:57:07   Log-Likelihood:                -51.061
 No. Observations:                  20   AIC:                             106.1
 Df Residuals:                      18   BIC:                             108.1
 Df Model:                           1                                         
 Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
 ----------------------------

<h4>La ecuación de la regresión lineal es entonces:</h4>
<strong>y=5.79+1.76x</strong>